site/src/pages/release-notes/1.25.0.mdx

```diff
@@ -0,0 +1,147 @@
+---
+date: 2023-08-22
+---
+
+## 🌟 New features
+
+- **GraalVM Support**: Users can now easily build [GraalVM](https://www.graalvm.org/) native images for Armeria applications. #5005
+- **Micrometer Observation Support**: Support for [Micrometer Observation](https://micrometer.io/docs/observation) is added.
+  Refer to <type://ObservationClient> or <type://ObservationService> for details on how to integrate with Armeria.  #4659 #4980
+  ```java
+  ObservationRegistry observationRegistry = ...
+  WebClient.builder()
+    .decorator(ObservationClient.newDecorator(observationRegistry))
+  ...
+  Server.builder()
+    .decorator(ObservationService.newDecorator(observationRegistry))
+  ...
+  ```
+- **WebSocket Client Support**: You can now send and receive data over [WebSocket](https://en.wikipedia.org/wiki/WebSocket)
+  using <type://WebSocketClient>. #4972
+  ```java
+  WebSocketClient client = WebSocketClient.of("ws://...");
+  client.connect("/").thenAccept(webSocketSession -> {
+    WebSocketWriter writer = WebSocket.streaming();
+    webSocketSessions.setOutbound(writer);
+    outbound.write("Hello!");
+
+    Subscriber<WebSocketFrame> subscriber = new Subscriber<WebSocketFrame>() {
+      ...
+    }
+    webSocketSessions.inbound().subscribe(subscriber);
+  });
+  ```
+- **Implement gRPC Richer Error Model more easily**: You can now easily use gRPC
+  [Richer Error Model](https://grpc.io/docs/guides/error/#richer-error-model) via <type://GoogleGrpcStatusFunction>. #4614 #4986
+  ```java
+  GoogleGrpcStatusFunction statusFunction = (ctx, throwable, metadata) -> {
+    if (throwable instanceof MyException) {
+      return com.google.rpc.Status.newBuilder()
+        .setCode(Code.UNAUTHENTICATED.getNumber())
+        .addDetails(detail(throwable))
+        .build();
+    }
+  };
+  Server.builder().service(
+    GrpcService.builder()
+      .exceptionMapping(statusFunction)
+  ```
+- **New API for Multipart Headers**: You can now retrieve headers from a multipart request in an annotated service
+  using <type://MultipartFile#headers()>.  #5106
+- **Access RequestLogProperty values more easily**: <type://RequestLogAccess#getIfAvailable()>
+  has been introduced, which allows users to access a <type://RequestLogProperty> immediately if available. #4956 #4966
+- **Keep an Idle Connection Alive on PING**: The `keepAliveOnPing` option has been introduced. Enabling this option will keep
+  an idle connection alive when a HTTP/2 PING frame or `OPTIONS * HTTP/1.1` is received. The option can be configured
+  by <type://ServerBuilder#idleTimeout(Duration,boolean)> or <type://ClientFactoryBuilder#idleTimeout(Duration,boolean)>. #4794 #4806
+- **Create a StreamMessage from Future**: You can now easily create a <type://StreamMessage> from a <type://CompletionStage>
```

Comment:
```suggestion
- **Create a StreamMessage from Future**: You can now easily create a <type://StreamMessage> from a `CompletionStage`
```
Because `CompletionStage` is not an API from Armeria, it's not rendered via `type`.